src/api/schema/log_events/mod.rs

```diff
@@ -0,0 +1,123 @@
+mod event;
+mod notification;
+
+use event::LogEvent;
+
+use crate::api::{
+    tap::{TapController, TapNotification, TapResult, TapSink},
+    ControlSender,
+};
+use async_graphql::{validators::IntRange, Context, Subscription, Union};
+use futures::StreamExt;
+use itertools::Itertools;
+use tokio::{select, stream::Stream, sync::mpsc, time};
+
+#[derive(Union)]
+/// Log event result which can be a payload for log events, or an error
+pub enum LogEventResult {
+    /// Log event payload
+    LogEvent(event::LogEvent),
+    /// Log notification
+    Notification(notification::LogEventNotification),
+}
+
+/// Convert an `api::TapResult` to the equivalent GraphQL type.
+impl From<TapResult> for LogEventResult {
+    fn from(t: TapResult) -> Self {
+        use notification::{LogEventNotification, LogEventNotificationType};
+
+        match t {
+            TapResult::LogEvent(name, ev) => Self::LogEvent(LogEvent::new(&name, ev)),
+            TapResult::Notification(name, n) => match n {
+                TapNotification::ComponentMatched => Self::Notification(LogEventNotification::new(
+                    &name,
+                    LogEventNotificationType::ComponentMatched,
+                )),
+                TapNotification::ComponentNotMatched => Self::Notification(
+                    LogEventNotification::new(&name, LogEventNotificationType::ComponentNotMatched),
+                ),
+            },
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct LogEventsSubscription;
+
+#[Subscription]
+impl LogEventsSubscription {
+    /// A stream of log events emitted from matched component(s)
+    pub async fn output_log_events<'a>(
+        &'a self,
+        ctx: &'a Context<'a>,
+        component_names: Vec<String>,
+        #[graphql(default = 500)] interval: i32,
+        #[graphql(default = 100, validator(IntRange(min = "1", max = "10_000")))] limit: usize,
```

Comment:
This allows for observing log events in batch sizes of up to 10,000. This feels a little large to me (it's probably quite expensive to unwind 10k records on the client, at default intervals of 0.5 seconds -- not to mention collect/serialize a JSON payload of that size in Vector) -- but we can adjust if needed.